R/create_monthly_costs.R

```diff
@@ -19,16 +20,57 @@ create_monthly_costs <- function(data,
   costs <- data %>%
     dplyr::select(dplyr::ends_with("_beddays")) %>%
     dplyr::rename_with(~ stringr::str_replace(., "_beddays", "_cost"))
+  # Fix the instances where the episode is a daycase;
+  # these will sometimes have 0.33 for the yearstay,
+  # this should be applied to the relevant month.
+  costs_daycase <- data %>%
+    dplyr::select(-dplyr::ends_with("_beddays")) %>%
+    dplyr::mutate(
+      daycase_added = (.data$record_keydate1 == .data$record_keydate2)
+    ) %>%
+    dplyr::mutate(daycase_check = .data$daycase_added) %>%
+    dplyr::mutate(whichmonth = dplyr::if_else(
+      .data$daycase_added,
+      lubridate::month(data$record_keydate1),
+      NA
+    )) %>%
+    dplyr::mutate(
+      whichmonth = month.abb[.data$whichmonth] %>%
+        tolower() %>%
+        paste0("_cost"),
+      daycase_added = dplyr::if_else(.data$daycase_added, 1, 0)
+    ) %>%
+    tidyr::pivot_wider(
+      names_from = .data$whichmonth,
+      values_from = .data$daycase_added,
+      values_fill = 0
+    ) %>%
```

Comment:
We can do the rename here which tidies up the above code.
```suggestion
    tidyr::pivot_wider(
      names_from = "whichmonth",
      names_glue = "{whichmonth}_cost",
      values_from = "daycase_added",
      values_fill = 0
    ) %>%
```

Reply:
Hi James. Sorry I have to revert this change. Actually, the variable `whichmonth` does not contain all 12 months. Hence, it will be short of some columns, such as `apr_cost`, if we use `name_glue`. However, the old bit of the pivot_wider code strangely creates all columns for 12 months. I guess it is because `month.abb[]` in fact creates a vector with levels being 12 months, and all levels will be created as columns.

Feel free to discuss the unelegant code @_@.